mautic/mautic
# Add send custom email to contact by API

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | "features"
| Bug fix?                               | no
| New feature?                           | yes
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | no
| Related user documentation PR URL      | 
| Related developer documentation PR URL | https://github.com/mautic/developer-documentation/pull/190
| Issue(s) addressed                     | 




#### Description:

Add support to API send custom content to contact



#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Setup API

Test by API

$params = [];
$params['fromEmail'] =  'test@test.com';
$params['fromName'] =  'Test tester';
$params['subject'] = 'xxx';
$params['content'] = 'custom content';

```
$api = new \Mautic\Api\Api($auth, $apiUrl);
$response = $api->makeRequest('emails/contact/{$contactId}/send/custom', $params, 'POST');
```

Test with API library after patch https://github.com/mautic/api-library/pull/241:

```
$api = new \Mautic\MauticApi($auth, $apiUrl);
$response = $emailsApi->sendCustomToContact($contactId, $params);
```

Both should send email
